stdlib/bigarray.mli

```diff
@@ -1050,3 +1050,47 @@ val reshape_3 :
   ('a, 'b, 'c) Genarray.t -> int -> int -> int -> ('a, 'b, 'c) Array3.t
 (** Specialized version of {!Bigarray.reshape} for reshaping to
    three-dimensional arrays. *)
+
+(** {1:bigarray_concurrency Bigarrays and concurrency safety}
+
+    Reading and writing bigarrays is safe in OCaml, even in presence of data
+    races.
+
+    Unfortunately, C does not offer such guaranteed and any data races anywhere
+    in a C program make the whole program unsafe.
+    Consequently, if bigarrays are used on the C side through the OCaml FFI,
+    data races must be avoided at any cost.
+
+    However, even within OCaml code, it is advised to avoid data races when
+    possible. Data races not only make execution non-deterministic, they also
+    considerably complicate the possible behaviors of programs.
+
+    A general caveat in presence of OCaml data races, concurrent array
+    operations might be interleaved leading to results that cannot be obtained
+    with a sequential execution of those operations.
+    For instance, at the end of
+{[open Bigarray
+let size = 100_000_000
+let a = Array1.init Int C_layout size (fun _ -> 1)
+let update f a () =
+  for i = 0 to size - 1 do a.{i} <- f a.{i} done
+let d1 = Domain.spawn (update (fun x -> x + 1) a)
+let d2 = Domain.spawn (update (fun x -> 2 * x + 1) a)
+let () = Domain.join d1; Domain.join d2
+]}
+    the bigarray [a] may contain a non-deterministic mixture of [2], [3], [4],
+    and [5] values.
+
+    Bigarrays have another caveat in presence of data races, concurrent bigarray
```

Comment:
"Bigarrays have another caveat in presence of data races," -> replace comma with colon